com.unity.multiplayer.mlapi/Tests/Editor/InterestTests.cs

```diff
@@ -0,0 +1,344 @@
+                nodes[i].NetworkObjectId = (ulong)(i + 100);
+                nm.SpawnManager.SpawnNetworkObjectLocally(nodes[i]);
+            }
+
+            nc.PlayerObject = MakeGameInterestObjectHelper(new Vector3(0.0f, 0.0f, 0.0f), null);
+
+            results.Clear();
+            nc.PlayerObject.NetworkObjectId = 100;
+            nm.InterestManager.QueryFor(nc, results);
+            Assert.True(results.Count - objectsBeforeAdd == numNodes / 2);
+
+            results.Clear();
+            nc.PlayerObject.NetworkObjectId = 101;
+            nm.InterestManager.QueryFor(nc, results);
+            Assert.True(results.Count - objectsBeforeAdd == numNodes / 2);
+
+            // now re-number the nodes
+            for (var i = 0; i < numNodes; i++)
+            {
+                nodes[i].NetworkObjectId = (ulong)(2 * i + 101); // 101, 103, 105, 107
+            }
+
+            results.Clear();
+            nc.PlayerObject.NetworkObjectId = 100;
+            nm.InterestManager.QueryFor(nc, results);
+            Assert.True(results.Count - objectsBeforeAdd == numNodes / 2); // should be unchanged
+
+            results.Clear();
+            nc.PlayerObject.NetworkObjectId = 101;
+            nm.InterestManager.QueryFor(nc, results);
+            Assert.True(results.Count - objectsBeforeAdd == numNodes / 2); // should be unchanged
+
+            // now update the nodes
+            for (var i = 0; i < numNodes; i++)
+            {
+                nodes[i].UpdateInterest();
+            }
+
+            results.Clear();
+            nc.PlayerObject.NetworkObjectId = 100;
+            nm.InterestManager.QueryFor(nc, results);
+            Assert.True(results.Count - objectsBeforeAdd == 0); // now zero, all are odd
+
+            results.Clear();
+            nc.PlayerObject.NetworkObjectId = 101;
+            nm.InterestManager.QueryFor(nc, results);
+            Assert.True(results.Count - objectsBeforeAdd == nodes.Length); // now 4, all are odd
+        }
+
+        [Test]
+        // Start is called before the first frame update
+        public void InterestRadiusCheck()
+        {
+            var nm = SetUpNetworkingManager();
+
+            var naiveRadiusNode = ScriptableObject.CreateInstance<InterestNode>();
+            var ris = ScriptableObject.CreateInstance<RadiusInterestStorage>();
+            ris.Radius = 1.5f;
+            naiveRadiusNode.InterestObjectStorage = ris;
+            var staticNode = ScriptableObject.CreateInstance<InterestNodeStatic>();
+
+            var results = new HashSet<NetworkObject>();
+            var nc = new NetworkClient()
+            {
+                ClientId = 1,
+            };
+
+            nm.InterestManager.QueryFor(nc, results);
+            int objectsBeforeAdd = results.Count;
+
+            var ok1 = MakeGameInterestObjectHelper(new Vector3(0.5f, 0.0f, 0.0f), naiveRadiusNode);
+            nm.SpawnManager.SpawnNetworkObjectLocally(ok1);
```

Comment:
I'd be interested to see this work with already spawned objects, moving out of radius.